Classes/Private/Manager/Submanagers/OCTSubmanagerFriends.m

```diff
@@ -211,6 +211,15 @@ - (void)tox:(OCTTox *)tox friendConnectionStatusChanged:(OCTToxConnectionStatus)
     [realmManager updateObject:friend withBlock:^(OCTFriend *theFriend) {
         theFriend.isConnected = (status != OCTToxConnectionStatusNone);
         theFriend.connectionStatus = status;
+
+        if (! theFriend.isConnected) {
+            NSDate *dateOffline = [tox friendGetLastOnlineWithFriendNumber:friendNumber error:nil];
+            NSTimeInterval timeSince = [dateOffline timeIntervalSince1970];
+            theFriend.lastSeenOnlineInterval = timeSince;
+        }
+        else {
+            theFriend.lastSeenOnlineInterval = 0;
```

Comment:
Hm, I don't get it. Why do you want to set `lastSeenOnlineInterval` to 0? It should have some value if friend was online at least once.

I see following cases:

0. If friend is online, we don't care about `lastSeenOnlineInterval`. Our documentation says that it `Contains actual information in case if friend has connectionStatus offline.`
1. Friend was online, now he goes offline. We should update `lastSeenOnlineInterval` with new value.
2. Friend was online, the app was killed. There was no chance of updating `lastSeenOnlineInterval` value for us. In that case we can update interval on next launch.

So, here you can leave only:

```objc
if (! theFriend.isConnected) {
    NSDate *dateOffline = [tox friendGetLastOnlineWithFriendNumber:friendNumber error:nil];
    theFriend.lastSeenOnlineInterval = [dateOffline timeIntervalSince1970];
}
```

And in `configure` method it would be also nice to update interval:

```objc
- (void)configure
{
    ...

     // update some of friend properties
    [realmManager updateObjectsWithoutNotification:^{
        for (OCTFriend *friend in friendsArray) {
            friend.status = OCTToxUserStatusNone;
            friend.isConnected = NO;
            friend.connectionStatus = OCTToxConnectionStatusNone;
            friend.isTyping = NO;
            NSDate *dateOffline = [tox friendGetLastOnlineWithFriendNumber:friendNumber error:nil];
            friend.lastSeenOnlineInterval = [dateOffline timeIntervalSince1970];
        }
    }];

    ...
}